.github/workflows/CI.yml

```diff
@@ -36,63 +43,75 @@ jobs:
     - name: Build
       working-directory: build
       run: cmake --build . -j $env:NUMBER_OF_PROCESSORS --config ${{ matrix.configuration }}
-    - uses: actions/upload-artifact@v1
+    - uses: actions/upload-artifact@v2
       if: matrix.toolset == 'v141_xp'
       with:
         name: openag-${{ runner.os }}-${{ matrix.configuration }}
         path: build\${{ matrix.configuration }}\client.dll
+        if-no-files-found: error
 
   build-linux-gcc:
-    runs-on: ubuntu-18.04
-    env:
-      CC: gcc-${{ matrix.gcc-ver }}
-      CXX: g++-${{ matrix.gcc-ver }}
+    name: build-linux (gcc-${{ matrix.gcc-ver }}, ${{ matrix.configuration }})
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        gcc-ver: [6, 7, 8]
+        gcc-ver: [6, 7, 8, 9, 10]
         configuration: [Release, Debug]
+    env:
+      CC: gcc-${{ matrix.gcc-ver }}
+      CXX: g++-${{ matrix.gcc-ver }}
     steps:
     - uses: actions/checkout@v2
       with:
         submodules: recursive
     - name: Setup
       run: |
-        sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+        if [ '${{ matrix.gcc-ver }}' = '6' ]; then
+          sudo add-apt-repository 'deb http://azure.archive.ubuntu.com/ubuntu/ bionic main'
+          sudo add-apt-repository 'deb http://azure.archive.ubuntu.com/ubuntu/ bionic universe'
+          sudo add-apt-repository 'deb http://azure.archive.ubuntu.com/ubuntu/ bionic-updates main'
+          sudo add-apt-repository 'deb http://azure.archive.ubuntu.com/ubuntu/ bionic-updates universe'
+        fi
         sudo apt-get update
-        sudo apt-get -y install g++-${{ matrix.gcc-ver }} g++-${{ matrix.gcc-ver }}-multilib linux-libc-dev-i386-cross mesa-common-dev ninja-build rapidjson-dev
+        sudo apt-get install -y g++-${{ matrix.gcc-ver }}-multilib libgl-dev ninja-build rapidjson-dev
         mkdir build
     - name: CMake generate
       working-directory: build
       run: cmake -G Ninja -DCMAKE_BUILD_TYPE=${{ matrix.configuration }} ..
     - name: Build
       working-directory: build
       run: ninja
-    - uses: actions/upload-artifact@v1
-      if: matrix.gcc-ver == '8'
+    - uses: actions/upload-artifact@v2
+      if: matrix.gcc-ver == '10'
       with:
         name: openag-${{ runner.os }}-${{ matrix.configuration }}
         path: build/client.so
+        if-no-files-found: error
 
   build-linux-clang:
-    runs-on: ubuntu-18.04
-    env:
-      compiler: clang
-      CC: clang
-      CXX: clang++
+    name: build-linux (clang-${{ matrix.clang-ver }}, ${{ matrix.configuration }})
+    runs-on: ubuntu-latest
```

Comment:
Also needs to be `ubuntu-18.04` or `20.04` since artifacts aren't uploaded from this.